packages/fxa-payments-server/src/routes/Product/index.tsx

```diff
@@ -275,7 +273,10 @@ export const Product = ({
     // if desired product is already subscribed to, show iap already subscribed error
     // else, product is not subscribed to, but on same product set/might be eligible for upgrade
     // show iap upgrade contact support error messaging
-    if (planUpgradeEligibility === 'blocked_iap') {
+    if (
+      (subscriptionChangeEligibility.result as any).eligibility ===
```

Comment:
Once we fix the type, as mentioned in a separate comment, this shouldn't be necessary.